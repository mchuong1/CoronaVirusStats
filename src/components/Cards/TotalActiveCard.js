import React from "react";
import active from '../../img/Active.png'
import { addComma } from '../../resource/Common'

class TotalActiveCard extends React.Component {

  render() {
    var { totalActive, newActive } = this.props
    newActive = addComma(newActive)
    totalActive = addComma(totalActive)
    return (
      <div className="card-icon red">
        <img alt="" src={active} className="icon"></img>
        <div className="title grid-column-2">Total Active</div>
        <span className="count grid-column-2" style={{ color: 'red' }}>{totalActive}</span>
        <div className="newcount grid-column-2">{newActive} New Active</div>
      </div>
    )
  }

  componentDidMount() { }
}

export default TotalActiveCard;
