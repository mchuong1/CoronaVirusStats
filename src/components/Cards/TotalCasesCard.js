import React from "react";
import virus from '../../img/virus.png'

class TotalCasesCard extends React.Component {

  render() {
    var { data } = this.props
    var totalConfirmed
    var newConfirmed
    if(data[1]){
      totalConfirmed = data[data.length-1].Confirmed
      newConfirmed = totalConfirmed - data[data.length-2].Confirmed
    }
    return (
        <div className="card orange">
          <img alt="" src={virus} className="icon"></img>
          <div className="title">Total Cases Confirmed</div>
          <span className="count" style={{color: "rgba(255,128,0)"}}>{totalConfirmed}</span>
          <div className="newcount">{newConfirmed} New Cases</div>
        </div>
    )
  }

  componentDidMount() {}
}

export default TotalCasesCard;
