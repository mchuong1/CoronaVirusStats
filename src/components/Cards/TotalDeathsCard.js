import React from "react";
import rip from '../../img/rip.png'

class TotalDeathsCard extends React.Component {

  render() {
    var { data } = this.props
    var totalDeaths
    var newDeaths
    if(data[1]){
      totalDeaths = data[data.length-1].Deaths
      newDeaths = totalDeaths - data[data.length-2].Deaths
    }
    return (
        <div className="card darkgreen">
          <img alt="" src={rip} className="icon"></img>
          <div className="title">Total Deaths</div>
          <span className="count" style={{color: "#085f63"}}>{totalDeaths}</span>
          <div className="newcount">{newDeaths} New Deaths</div>
        </div>
    )
  }

  componentDidMount() {}
}

export default TotalDeathsCard;
