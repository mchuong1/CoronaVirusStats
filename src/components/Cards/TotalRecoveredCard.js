import React from "react";
import recovered from '../../img/recovered.png'

class TotalRecoveredCard extends React.Component {

  render() {
    var { data } = this.props
    var totalRecovered
    var newRecovered
    if(data[1]){
      totalRecovered = data[data.length-1].Recovered
      newRecovered = totalRecovered - data[data.length-2].Recovered
    }
    return (
        <div className="card green">
          <img alt="" src={recovered} className="icon"></img>
          <div className="title">Total Recovered</div>
          <span className="count" style={{color: "rgba(0,204,0)"}}>{totalRecovered}</span>
          <div className="newcount">{newRecovered} New Recovered</div>
        </div>
    )
  }

  componentDidMount() {}
}

export default TotalRecoveredCard;
