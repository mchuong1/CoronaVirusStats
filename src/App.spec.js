import React from 'react'
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import TotalCasesCard from './components/Cards/TotalCasesCard';
import TotalDeathsCard from './components/Cards/TotalDeathsCard';
import TotalRecoveredCard from './components/Cards/TotalRecoveredCard';
import TotalActiveCard from './components/Cards/TotalActiveCard';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

var stubdata = [
  {
    Active: 1,
    City: "",
    CityCode: "",
    Confirmed: 1,
    Country: "United States of America",
    CountryCode: "",
    Date: "2020-01-22T00:00:00Z",
    Deaths: 0,
    Lat: "0",
    Lon: "0",
    Province: "",
    Recovered: 0
  }, 
  {
    Active: 1,
    City: "",
    CityCode: "",
    Confirmed: 1,
    Country: "United States of America",
    CountryCode: "",
    Date: "2020-01-23T00:00:00Z",
    Deaths: 0,
    Lat: "0",
    Lon: "0",
    Province: "",
    Recovered: 0
  }]

it("TotalCasesCard renders with a number", () => {
  act(() => {
    render(<TotalCasesCard totalConfirmed={1} newConfirmed={0}/>, container);
  });
  expect(container.textContent).toBe("Total Confirmed10 New Cases");
});

it("TotalActiveCard renders with a number", () => {
  act(() => {
    render(<TotalActiveCard totalActive={0} newActive={0}/>, container);
  });
  expect(container.textContent).toBe("Total Active00 New Active");
})

it("TotalRecoveredeCard renders with a number", () => {
  act(() => {
    render(<TotalRecoveredCard totalRecovered={0} newRecovered={0}/>, container);
  });
  expect(container.textContent).toBe("Total Recovered00 New Recovered");
});

it("TotalDeathsCard renders with a number", () =>{
  act(()=> {
    render(<TotalDeathsCard totalDeaths={0} newDeaths={0}/>, container);
  });
  expect(container.textContent).toBe("Total Deaths00 New Deaths")
})

