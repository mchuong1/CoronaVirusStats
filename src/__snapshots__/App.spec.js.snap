// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveGraph renders correctly 1`] = `
<div
  className="graph red"
  id="Active"
>
  <h3
    className="title-graph"
  >
    Daily Active Cases
  </h3>
</div>
`;

exports[`CountryList renders correctly 1`] = `
<div
  className="autocomplete"
>
  <input
    onChange={[Function]}
    placeholder="Enter a country..."
    type="text"
    value=""
  />
</div>
`;

exports[`DeathGraph renders correctly 1`] = `
<div
  className="graph darkgreen"
  id="Deaths"
>
  <h3
    className="title-graph"
  >
    Daily Death Cases
  </h3>
</div>
`;

exports[`NewCasesGraph renders correctly 1`] = `
<div
  className="graph orange"
  id="New"
>
  <h3
    className="title-graph"
  >
    Daily New Cases
  </h3>
</div>
`;

exports[`RecoveredGraph renders correctly 1`] = `
<div
  className="graph green"
  id="Recovered"
>
  <h3
    className="title-graph"
  >
    Daily Recovered Cases
  </h3>
</div>
`;

exports[`SideNavBar renders correctly 1`] = `
<nav>
  <img
    alt=""
    className="menu"
    id="menu"
    onClick={[Function]}
    src="bars.png"
  />
  <div
    className="navbar"
    id="navbar"
  >
    <p
      id="close"
      onClick={[Function]}
    >
      X
    </p>
    <h2>
      Coronavirus Stats
    </h2>
    <ul>
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <li>
          World
        </li>
      </a>
      <a
        href="/Country"
        onClick={[Function]}
        style={
          Object {
            "color": "black",
            "textDecoration": "none",
          }
        }
      >
        <li>
          Lookup by Country
        </li>
      </a>
    </ul>
    <div
      className="footer"
    >
      <p>
        Made with 
        <a
          href="https://covid19api.com/"
        >
          COVID19 API
        </a>
         and 
        <a
          href="https://www.chartjs.org/"
        >
          ChartJs
        </a>
      </p>
    </div>
  </div>
</nav>
`;

exports[`Tab renders correctly 1`] = `
<div
  className="tab"
>
  <ul>
    <a
      className="tabitem"
      id="alltab"
      onClick={[Function]}
    >
      All
    </a>
    <a
      className="tabitem"
      onClick={[Function]}
    >
      New
    </a>
    <a
      className="tabitem"
      onClick={[Function]}
    >
      Active
    </a>
    <a
      className="tabitem"
      onClick={[Function]}
    >
      Recovered
    </a>
    <a
      className="tabitem"
      id="deathtab"
      onClick={[Function]}
    >
      Deaths
    </a>
  </ul>
</div>
`;

exports[`TotalActiveCard renders correctly 1`] = `
<div
  className="card red"
>
  <img
    alt=""
    className="icon"
    src="Active.png"
  />
  <div
    className="title"
  >
    Total Active
  </div>
  <span
    className="count"
    style={
      Object {
        "color": "red",
      }
    }
  >
    0
  </span>
  <div
    className="newcount"
  >
    0
     New Active
  </div>
</div>
`;

exports[`TotalConfirmedCard renders correctly 1`] = `
<div
  className="card orange"
>
  <img
    alt=""
    className="icon"
    src="virus.png"
  />
  <div
    className="title"
  >
    Total Confirmed
  </div>
  <span
    className="count"
    style={
      Object {
        "color": "rgba(255,128,0)",
      }
    }
  >
    1
  </span>
  <div
    className="newcount"
  >
    0
     New Cases
  </div>
</div>
`;

exports[`TotalDeathsCard renders correctly 1`] = `
<div
  className="card darkgreen"
>
  <img
    alt=""
    className="icon"
    src="rip.png"
  />
  <div
    className="title"
  >
    Total Deaths
  </div>
  <span
    className="count"
    style={
      Object {
        "color": "#085f63",
      }
    }
  >
    0
  </span>
  <div
    className="newcount"
  >
    0
     New Deaths
  </div>
</div>
`;

exports[`TotalRecoveredCard renders correctly 1`] = `
<div
  className="card green"
>
  <img
    alt=""
    className="icon"
    src="recovered.png"
  />
  <div
    className="title"
  >
    Total Recovered
  </div>
  <span
    className="count"
    style={
      Object {
        "color": "rgba(0,204,0)",
      }
    }
  >
    0
  </span>
  <div
    className="newcount"
  >
    0
     New Recovered
  </div>
</div>
`;

exports[`World renders correctly 1`] = `
<div
  className="world"
>
  <div
    className="header"
  >
    <h1>
      World Statistics
    </h1>
  </div>
  <div
    className="world-card-row"
  >
    <div
      className="card orange"
    >
      <img
        alt=""
        className="icon"
        src="virus.png"
      />
      <div
        className="title"
      >
        Total Confirmed
      </div>
      <span
        className="count"
        style={
          Object {
            "color": "rgba(255,128,0)",
          }
        }
      >
        
      </span>
      <div
        className="newcount"
      >
        
         New Cases
      </div>
    </div>
    <div
      className="card green"
    >
      <img
        alt=""
        className="icon"
        src="recovered.png"
      />
      <div
        className="title"
      >
        Total Recovered
      </div>
      <span
        className="count"
        style={
          Object {
            "color": "rgba(0,204,0)",
          }
        }
      >
        
      </span>
      <div
        className="newcount"
      >
        
         New Recovered
      </div>
    </div>
    <div
      className="card darkgreen"
    >
      <img
        alt=""
        className="icon"
        src="rip.png"
      />
      <div
        className="title"
      >
        Total Deaths
      </div>
      <span
        className="count"
        style={
          Object {
            "color": "#085f63",
          }
        }
      >
        undefined
      </span>
      <div
        className="newcount"
      >
        undefined
         New Deaths
      </div>
    </div>
  </div>
</div>
`;
